app/models/partner.rb

```diff
@@ -312,6 +312,28 @@ def owned_neighbourhood_ids
     neighbourhood_ids
   end
 
+  # returns true if all
+  #  - address in user neighbourhood set
+  #  - service areas in user neighbourhood set
+  #  - tags in users partnership tag set
+  def owned_exclusively_by?(user)
```

Comment:
This is a method that is really asking if a user is allowed to access a resource, and given that all the other methods in the scope (ie. `.admin_for_partner?`) reside in the User model, I think it would make a lot more sense if this did too? 

If you have a look at the existing method `def only_neighbourhood_admin_for_partner?` we could also add a simple `def only_partnership_admin_for_partner?` method which could be checked first. Then the logic hierarchy would be clearer - partner admins and roots get the most control, followed by partnership admins, followed by neighbourhood admins.

In that method you would need to then check if a user is a `partnership_admin`, and then run the logic you've implemented here.

You'll also need to modify `only_neighbourhood_admin_for_partner` to check first if the user is a `partnership_admin || neighbourhood_admin` and not just a neighbourhood admin.